temporian/core/data/node.py

```diff
@@ -131,7 +307,8 @@ def __ne__(self, other: Any) -> Node:
 
             return not_equal_scalar(input=self, value=other)
 
-        self._nope("compare", other, "(int,float,bool,str)")
+        self._raise_error("ne", other, "int,float,bool,str")
+        assert False
```

Comment:
what do these assertions provide after already raising an error?

Reply:
Maybe ignoring pylint's warning would be clearer for developers? The assert confused me at first

Reply:
I would argue that we don't want any pylint warning in the code ultimately :), and I am hopping we can enable this as a check in github pre-submit. The alternative is to disable pylint warning locally, but the assert feels more robust.